### DISCUSSION TITLE
Q: 在项目中引入 waline，是否会受到 GPL 协议传播 || Q: Will introducing waline into the project be affected by the GPL license?

### DISCUSSION THREAD

**Discussion by AnzhiZhang** [walinejs/waline#2499] [Q&A] [answered]
### Describe the bug | 问题描述

想咨询一下在项目中引入 waline，是否会受到 GPL 协议传播，即引入 waline 的整个项目也需要以相同协议开源，还是只有对 waline 本身做修改的项目会受到限制。

### Website URL | 问题网站

.

### Where your waline deploy? | 服务部署在哪里？

Vercel (Default)

### Where your comment data store? | 数据存储在哪里？

LeanCloud(https://leancloud.app)

---
### Describe the bug | Problem description

I would like to inquire whether the introduction of waline into the project will be subject to the GPL license. That is, the entire project that introduces waline needs to be open sourced under the same license, or will only projects that modify waline itself be restricted.

### Website URL | Problem website

.

### Where your waline deploy? | Where is the service deployed?

Vercel (Default)

### Where your comment data store? | Where is the data stored?

LeanCloud(https://leancloud.app)

**Comment by lizheming** [comment, marked as answer]
如果你只是纯使用的话，是不需要按照 GPL 开源的。只有你用了这个程序，然后还要分发的话，才会有 GPL 的限制。

---
If you just use it purely, you don't need to open source it under the GPL. Only if you use this program and then distribute it will there be restrictions under the GPL.

**Reply by AnzhiZhang** [reply]
哪怕是作为第三方包引入，需要分发的话，也要使用 GPL 开源吗

---
Even if it is introduced as a third-party package, if it needs to be distributed, should it be open source using GPL?

**Reply by lizheming** [reply]
理论上是的。如果有觉得不合适，可以说下你的场景，我看下是否需要修改协议。Waline 使用 GPL 的原因是因为 Waline 的后台样式源自 Typecho，Typecho 是 GPL 协议所以整体项目的代码就被传染成 GPL 了。

---
Theoretically yes. If you feel it's inappropriate, you can tell me your scenario and I'll see if the agreement needs to be modified. The reason why Waline uses GPL is because Waline's backend style is derived from Typecho. Typecho is a GPL protocol, so the entire project code is infected with GPL.

**Reply by AnzhiZhang** [reply]
不用了，谢谢解答，我是提供网络应用，不需要二次分发，只是好奇问一下

---
No, thank you for your answer. I am providing a network application and do not need secondary distribution. I am just curious to ask
